+bisos/+iteration/+step/termination.m

```diff
@@ -0,0 +1,134 @@
+classdef termination < bisos.iteration.Step
+   
+properties
+    type = 'termination';
+    varout = [];
+    varin;
+
+    operator;
+    
+    % Structures with info about condition 
+    prev = struct('fhan', @(x) x, 'vars', []);
+    curr = struct('fhan', @(x) x, 'vars', []);
+end
+
+methods
+    function step = termination(~, previous, current, op, varargin)
+        % setup termination rule
+        
+        step.curr = step.setup2struct(step.curr, current);
+        step.varin = step.curr.vars;
+        step.prev = step.setup2struct(step.prev, previous);
+        
+        % save operator (maybe I should process it for any kind that 
+        % may appear)
+        
+        switch (op)
+        case {'>=', 'ge'} 
+           step.operator = @ge; 
+        case {'<=', 'le'}
+           step.operator = @le; 
+        case {'<', 'lt'}
+           step.operator = @lt; 
+        case {'>', 'gt'}
+           step.operator = @gt;  
+        case {'==', 'eq'}
+           step.operator = @eq;  
+        case {op, '~=', 'ne'}
+           step.operator = @ne;
+        otherwise
+           error('Operator "%s" not valid.', op)
+        end
+    end
+    
+    function [sol,info,stop] = run(step,~,info,sol,varargin)
+        % Run objective step.
+        stop = false;
+
+        if info.iter > 1
+            stepinfo = getinfo(step,info);
+        
+            % Prepare current term for condition evaluation    
+            value2 = step.calcfhan(sol, step.curr);
+        
+            % Get previous term for condition evaluation
+            value1 = stepinfo.value;
+
+            if step.operator(value1, value2)
+                info.converged = true;
+            end
+        end
+        
+        % Store 'previous' term for next iteration
+        stepinfo.value = step.calcfhan(sol, step.prev);
+        
+        info = setinfo(step,info,stepinfo);
+    end
+end
+
+methods (Static, Access=private)
+    function value = calcfhan(sol, sstruct) 
+        % sol: solutions that are accessed 
+        % sstruct: state struture, it can be either previous or current
+        % state
+
+        % preallocate auxiliary variable 'v'
+        v = cell(1,length(sstruct.vars));
+
+        for i=1:length(sstruct.vars)
+            if isfield(sol, sstruct.vars{i})
+                v{i} = double(sol.(sstruct.vars{i}));
+            else
+                error('Non existance of variable in solutions.');
```

Comment:
We don't need to check this here if you checked the variables in the constructor.